projects/repo/toolbox/notifications/slack/api.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import os
+import pathlib
+import requests
+
+from datetime import datetime
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+
+CHANNEL_ID = "C07NS5TAKPA"
+DEFAULT_MESSAGE = "🧵 Thread for PR #{}"
+
+
+def fetch_pr_creation_time(org: str, repo: str, pr_number: str):
+    """Fetch the PR creation time to filter out Slack messages.
+    No user_token is needed to list PRs of public repositories."""
```

Comment:
in the bare-metal clusters, with a single "internet exit point", we sometimes hit rate limits because of the rest of RH :/ 
We'll address that when we see it

Reply:
strange indeed
maybe you can add a `log.info(f"Trying to get the PR URL: {url}")`